### ISSUE TITLE
Calling `Cockpit::instance()` breaks error handling when using spaces/space urls

### ISSUE THREAD

**Issue by raffaelj** [Cockpit-HQ/Cockpit#131]
I'm experimenting with custom folder patterns and loading Cockpit as a library. It took me one and a half day to find out, why the error handling was broken and the execution always ended in a half rendered template after throwing an Exception. It turned out that a simple `t('Language')` in a nested template file was the root of all evil.

The reproduction of my issue is quite simple.

1. Create a test addon `addons/test/bootstrap.php`:

```php
<?php
$this->on('app.admin.request', function() {
    $this->bind('/broken', function() {
        return t('break error handler') . 1/0;
    });
});
```

2. Open `/broken` in a browser --> Error is caught with a debug page like expected
3. Create a space "test" and login
4. Open `/:test/broken` in a browser --> Empty page, no error handling

I didn't do more tests, but this should also affect `_t()` and `_allowed()`. I'm not sure about async calls with changed paths via config file, but they should be safe with calling `Cockpit::instance('{$envDir}')`.

Now I had a look at the `/path/to/lib-dir/cockpit/storage/data` folder and found all the missing error logs, that weren't printed. So the error event still triggers the system log, but in the wrong location.

**Comment by aheinze** [comment]
Hi Raffael 👋

Thanks for the finding and reporting 🙏 Should be fixed with this commit: https://github.com/Cockpit-HQ/Cockpit/commit/bfe69fd5171e62b634a7a138e12c9934226c966e

Can can retry your tests wit the latest dev release fro here: https://files.getcockpit.com/?dir=releases/develop

**Comment by aheinze** [comment]
Btw, I also would recommend to use shorthand functions only in views. In controllers etc I recommend to use the helpers on the app instance. So regarding your code, it would look like this:

```php
<?php

$this->on('app.admin.request', function() {
    $this->bind('/broken', function() {
        return $this->helper('i18n')->get('break error handler') . 1/0;
    });
});

```

**Comment by raffaelj** [comment]
> Can can retry your tests wit the latest dev release fro here: https://files.getcockpit.com/?dir=releases/develop

I prefer to `git pull` and mount the whole repo in a docker volume, but yes errors are caught and printed now.

They are also logged in the right location, so url `/broken` logs in folder `/storage/data` and `/:test/broken` logs in `/.spaces/test/storage/data`.

`return $this->helper('i18n')->get('break error handler') . 1/0;` defeats the purpose of breaking the exception handler ;-)

Spaces seem to work now.

I had a look at my experimental setup with custom folder patterns. After running `composer update` and adding `$GLOBALS['APP'] = $app;` after initialising Cockpit, my half rendered templates are showing up as error pages and the logs are created in the right location, too. Now I'm much happier.

This was an ugly one. Debugging a broken exception handler in a non-standard setup while setting up an error handler, that turns `E_WARNING`s into exceptions to throw errors if nested templates are missing - sucks. All my tests to reproduce the error in a simpler setup failed until I noticed that single translated string in a nested template. But I learned a lot about error handling and output buffering on that way, which is always good to know :-)

Thanks for the quick fix.
